README.md

```diff
@@ -0,0 +1,171 @@
+## Design proposal: process manager API
+
+This document outlines a design proposal for a prototype of a service API to manage Linux processes.
+
+There are four major components described below:
+ - Server
+ - Library
+ - API
+ - Client
+
+### Server
+
+The service is implemented as a server, allowing multiple concurrent client connections.
+
+The server provides the following functionality:
+ - exposes gRPC endpoint.
+ - leverages mTLS for client-server authentication.
+ - implements API proto spec and invokes corresponding calls in the core library.
+ - applies ressource control by adds itself to the beforehand configured cgroup.
+
+#### Authentication and authorization
+
+The authentication is implemented via mTLS. The server and the clients are using certificates signed by a common CA.
+
+Each client has an individual certificate with a unique Common Name (`CN`).
+
+If the server cannot authenticate the client, the request will be rejected with an error.
+
+The commands below could be used to generate certificates.
+
+
+```bash
+# create CA key pair and self signed certificate
+openssl req -x509 -newkey rsa:4096 -keyout ca.key -out ca.crt -days 365 -nodes -subj "/CN=RootCA" -config host.conf
+
+# create key pair and CSR for server and clients
+openssl genrsa -out server.key 2048
+openssl genrsa -out client1.key 2048
+openssl genrsa -out client2.key 2048
+
+openssl req -new -key server.key -subj "/CN=server" -out server.csr
+openssl req -new -key client1.key -subj "/CN=client1" -out client1.csr
+openssl req -new -key client2.key -subj "/CN=client2" -out client2.csr
+
+# sign CSR with CA
+openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 365 -extfile host-server.conf
+openssl x509 -req -in client1.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client1.crt -days 365
+openssl x509 -req -in client2.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client2.crt -days 365
+```
+
+#### Resource control
+
+The control over compute, network and memory resources is provided by means of cgroups.
```

Comment:
Network resource limit is not required in this challenge though Disk IO limit is missing.